alert-database/src/test/java/com/synopsys/integration/alert/database/api/DefaultNotificationManagerTest.java

```diff
@@ -0,0 +1,308 @@
+    private final String providerConfigName = "providerConfigName-test"; //provider.common.config.name
+    private final String notificationType = "notificationType-test";
+    private final String content = "content";
+
+    private final String KEY_PROVIDER_CONFIG_NAME = ProviderDescriptor.KEY_PROVIDER_CONFIG_NAME;
+    private final String fieldValue = "test-channel.common.name-value";
+
+    private AlertNotificationModel expectedAlertNotificationModel = new AlertNotificationModel(id, providerConfigId, provider, fieldValue, notificationType, content, new Date(), new Date());
+
+    @Test
+    public void saveAllNotificationsTest() throws Exception {
+        Date createdAt = new Date();
+        Date providerCreationTime = new Date(System.currentTimeMillis() - 10000);
+
+        AlertNotificationModel alertNotificationModel = new AlertNotificationModel(null, providerConfigId, provider, providerConfigName, notificationType, content, createdAt, providerCreationTime);
+        NotificationEntity notificationEntity = new NotificationEntity(id, createdAt, provider, providerConfigId, providerCreationTime, notificationType, content);
+        ConfigurationModel configurationModel = createConfigurationModel();
+
+        NotificationContentRepository notificationContentRepository = Mockito.mock(NotificationContentRepository.class);
+        ConfigurationAccessor configurationAccessor = Mockito.mock(ConfigurationAccessor.class);
+        EventManager eventManager = Mockito.mock(EventManager.class);
+
+        Mockito.when(notificationContentRepository.saveAll(Mockito.any())).thenReturn(List.of(notificationEntity));
+        Mockito.when(configurationAccessor.getConfigurationById(Mockito.any())).thenReturn(Optional.of(configurationModel));
+
+        DefaultNotificationManager notificationManager = new DefaultNotificationManager(notificationContentRepository, null, null, configurationAccessor, eventManager);
+        List<AlertNotificationModel> alertNotificationModelList = notificationManager.saveAllNotifications(List.of(alertNotificationModel));
+
+        assertEquals(1, alertNotificationModelList.size());
+        AlertNotificationModel testAlertNotificationModel = alertNotificationModelList.get(0);
+        testExpectedAlertNotificationModel(expectedAlertNotificationModel, testAlertNotificationModel);
+    }
+
+    @Test
+    public void saveAllNotificationsEmptyModelListTest() {
+        NotificationContentRepository notificationContentRepository = Mockito.mock(NotificationContentRepository.class);
+
+        Mockito.when(notificationContentRepository.saveAll(Mockito.any())).thenReturn(new ArrayList<>());
+
+        DefaultNotificationManager notificationManager = new DefaultNotificationManager(notificationContentRepository, null, null, null, null);
+        List<AlertNotificationModel> alertNotificationModelList = notificationManager.saveAllNotifications(new ArrayList<>());
+
+        assertEquals(0, alertNotificationModelList.size());
```

Comment:
Small thing, but you could just check the isEmpty method here.

Reply:
I have assertEquals 0 in several places when checking for empty lists, I can go through and change all of them to isEmpty if thats a better solution